### ISSUE TITLE
Support for other Doc Strings content types

### ISSUE THREAD

**Issue by bdo** [mapado/prettier-plugin-gherkin#20]
First and foremost, amazing plugin! 🤩

I had written some custom scripts to format our cucumber features on a project at work and was glad I could remove all the code and replace it with your plugin with nearly no changes! 🥳

Regarding [Doc Strings](https://cucumber.io/docs/gherkin/reference/#doc-strings) content types (mediaType), the [current handling](https://github.com/mapado/prettier-plugin-gherkin/blob/5d908953d7328994302182f3a9d159a0a288156f/src/index.ts#L639-L668) is already really good, since I think most developers would only be using JSON and XML.

However for projects using additional languages, it would be great if we could specify a [prettier parser](https://prettier.io/docs/en/options.html#parser) name as the mediaType! That way, we would be able to reformat Doc Strings for all languages that have a prettier parser.

This also mean we could use [Prettier for XML](https://github.com/prettier/plugin-xml) if we wanted to as an alternative for formatting XML (instead of the more limited html parser).

I'll throw a PR together as soon as possible to illustrate what I'm thinking of. 

What do you guys think?

**Comment by jdeniau** [comment]
Hi, 

First of all, no need for lobbying (9 reactions in 1 hour ?! 🤨 )

Thank you for your warn comment though 👍 

I see two possible evolution in your message:

### extend the embed parser to allow any type

We could use the docstring annotation to do that 
![image](https://github.com/mapado/prettier-plugin-gherkin/assets/1398469/566f6dec-c529-4e58-9c5a-f7aae30084ee)

_use the markdown parser here_

And fallback on xml and json.

### Use the xml parser plugin if available

We might try to parse as xml as default with something like this:

```ts
let xmlParsedContent;

try {
  xmlParsedContent = await textToDoc(content, { ...options, parser: 'xml' });
} catch (e) {
  // fallback to the html parser if the xml plugin is not available
  xmlParsedContent = await textToDoc(content, { ...options, parser: 'html' });
}

return [
  node.delimiter,
  mediaType,
  printHardline(),
  xmlParsedContent,
  printHardline(),
  node.delimiter,
];
```

**Comment by bdo** [comment]
Hi Julien,

Thanks for your detailed answer and for giving some thoughts into the problem!

You suggestion makes sense. However, attempting XML parsing on every file might be expensive. The JSON fallback does make sense though, assuming most docstrings are JSON in Node.js projects.

Now that my PR is up, we can probably continue the discussion on the PR?

PS: Fair comment about the lobbying, haha! That was my team here at Eurosport, hope it did not annoy you too much, if so, apologies. And sorry that I pushed my PR yesterday before checking the issue comments!

**Comment by jdeniau** [comment]
I merged your PR and released as `2.3.0`. Thank your for that, it's way better to support any language and not only JSON and XML.

Good to know that Eurosport is using our plugin !

**Comment by bdo** [comment]
Thanks so much Julien! ❤️
